test/DynamoCoreTests/PythonMigrationAssistantTests.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Dynamo.PythonMigration.MigrationAssistant;
+using NUnit.Framework;
+using System.Collections.Generic;
+
+namespace Dynamo.Tests
+{
+    public class PythonMigrationAssistantTests
+    {
+        [Test]
+        public void CanMigratePyton2CodeToPython3Code()
+        {
+            // Arrange
+            var expectedPython3Code = @"print(""Hello, Dynamo!"")";
+
+            var python2Code = @"print ""Hello, Dynamo!""";
+
+            // Act
+            var migratedScript = ScriptMigrator.MigrateCode(python2Code);
+
+            // Assert
+            Assert.AreNotEqual(python2Code, migratedScript);
+            Assert.AreEqual(expectedPython3Code, migratedScript);
+        }
+
+        [Test]
+        public void MigrationWillNotChangePython3CompatibleCode()
+        {
+            // Arrange
+            var expectedPython3Code = @"for x in range(1, 5): print(x)";
+
+            // Act
+            var migratedScript = ScriptMigrator.MigrateCode(expectedPython3Code);
+
+            // Assert
+            Assert.AreEqual(expectedPython3Code, migratedScript);
+        }
+
+        [Test]
+        public void MigrationWillNotChangeCodeAfterBeingConverted()
+        {
+            // Arrange
+            var originalPython2Code = @"for x in xrange(100): continue";
+            var expectedPython3CodeAfterMigration = @"for x in range(100): continue";
+
+            // Act
+            var migratedScript = ScriptMigrator.MigrateCode(originalPython2Code);
+            var migratedScript2 = ScriptMigrator.MigrateCode(migratedScript);
+
+            // Assert
+            Assert.AreEqual(expectedPython3CodeAfterMigration, migratedScript);
+            Assert.AreEqual(expectedPython3CodeAfterMigration, migratedScript2);
+        }
```

Comment:
By the name I understand the intention of this test, but I don't think it tests anything different than the previous test. We already know ScriptMigrator is calling a script and that it is stateless, so calling 2to3 many times does not seem a useful scenario.

Reply:
chiming in with a bit of context @mmisol for this test.

Because the `2to3` migration is an external (py) component that we don't really control, we wanted to make sure the user's experience of migrating a script is `idempotent`. While not useful in itself, it's more to address the edge cases where users might apply the migration multiple times to the same code, either 
- by accident
- because script came from someone else and they're unsure if it's Python 2 or 3

It also confirms that the external migration script doesn't miss anything if only run once.

I'd suggest we keep this test for the reasons above, but up to you folks :)

Reply:
@radumg Since you mentioned idempotency let's get mathematical :) The second test shows that `x = f(x)` when `x` doesn't have Python 2 in it. The third tests shows that `y = f(f(y))` with a `y` with the same characteristics as `x` in the previous test. The second test is already showing idempotency, therefore my argument to delete this third test.

Reply:
> @radumg let's get mathematical :)

haha, always love me some math🤓 

> @radumg The second test is already showing idempotency, therefore my argument to delete this third test.

You're totally right, hadn't looked at second test when I made the comment, thought we only had the normal migration test. Let's delete this one then since we all agree ? 😁